network-store-client/src/main/java/com/powsybl/network/store/client/PreloadingStrategy.java

```diff
@@ -1,16 +1,75 @@
 /**
- * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 package com.powsybl.network.store.client;
 
+import com.powsybl.network.store.model.ResourceType;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
 /**
- * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Joris Mancini <joris.mancini_externe at rte-france.com>
  */
-public enum PreloadingStrategy {
-    NONE,
-    COLLECTION,
-    ALL_COLLECTIONS_NEEDED_FOR_BUS_VIEW
+@Getter
+@Builder
+public class PreloadingStrategy {
+
+    @Builder.Default
+    private boolean ignored = false;
+
+    @Builder.Default
+    private boolean collection = false;
+
+    @Builder.Default
+    private List<PreloadingResource> resources = Collections.emptyList();
+
+    public List<ResourceType> getResourceTypes() {
+        return resources.stream().map(PreloadingResource::getType).toList();
+    }
+
+    public static PreloadingStrategy none() {
+        return PreloadingStrategy.builder().ignored(true).build();
```

Comment:
Not a very nice design IMHO to have 2 words "ignored" and "none" for the same thing

Reply:
Not very important, both work, but I think the previous design of having a NONE preloading strategy was better than ignored() or null